<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: Genetic Relationship Matrix (GRM) for SNP genotype data</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body>

<table width="100%" summary="page for snpgdsGRM {SNPRelate}"><tr><td>snpgdsGRM {SNPRelate}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>
Genetic Relationship Matrix (GRM) for SNP genotype data
</h2>

<h3>Description</h3>

<p>Calculate Genetic Relationship Matrix (GRM) using SNP genotype data.
</p>


<h3>Usage</h3>

<pre>
snpgdsGRM(gdsobj, sample.id=NULL, snp.id=NULL, autosome.only=TRUE,
    remove.monosnp=TRUE, maf=NaN, missing.rate=NaN,
    method=c("GCTA", "Eigenstrat", "EIGMIX", "Weighted", "Corr", "IndivBeta"),
    num.thread=1L, with.id=TRUE, verbose=TRUE)
</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>gdsobj</code></td>
<td>
<p>an object of class <code><a href="SNPGDSFileClass-class.html">SNPGDSFileClass</a></code>,
a SNP GDS file</p>
</td></tr>
<tr valign="top"><td><code>sample.id</code></td>
<td>
<p>a vector of sample id specifying selected samples;
if NULL, all samples are used</p>
</td></tr>
<tr valign="top"><td><code>snp.id</code></td>
<td>
<p>a vector of snp id specifying selected SNPs; if NULL,
all SNPs are used</p>
</td></tr>
<tr valign="top"><td><code>autosome.only</code></td>
<td>
<p>if <code>TRUE</code>, use autosomal SNPs only; if it is a
numeric or character value, keep SNPs according to the specified
chromosome</p>
</td></tr>
<tr valign="top"><td><code>remove.monosnp</code></td>
<td>
<p>if TRUE, remove monomorphic SNPs</p>
</td></tr>
<tr valign="top"><td><code>maf</code></td>
<td>
<p>to use the SNPs with &quot;&gt;= maf&quot; only; if NaN, no MAF threshold</p>
</td></tr>
<tr valign="top"><td><code>missing.rate</code></td>
<td>
<p>to use the SNPs with &quot;&lt;= missing.rate&quot; only; if NaN,
no missing threshold</p>
</td></tr>
<tr valign="top"><td><code>method</code></td>
<td>
<p>&quot;GCTA&quot; &ndash; genetic relationship matrix defined in CGTA;
&quot;Eigenstrat&quot; &ndash; genetic covariance matrix in EIGENSTRAT;
&quot;EIGMIX&quot; &ndash; two times coancestry matrix defined in Zheng &amp; Weir (2015),
&quot;Weighted&quot; &ndash; weighted GCTA, as the same as &quot;EIGMIX&quot;,
&quot;Corr&quot; &ndash; Scaled GCTA GRM (dividing each i,j element by the
product of the square root of the i,i and j,j elements),
&quot;IndivBeta&quot; &ndash; two times individual beta estimate; see details</p>
</td></tr>
<tr valign="top"><td><code>num.thread</code></td>
<td>
<p>the number of (CPU) cores used; if <code>NA</code>, detect
the number of cores automatically</p>
</td></tr>
<tr valign="top"><td><code>with.id</code></td>
<td>
<p>if <code>TRUE</code>, the returned value with <code>sample.id</code>
and <code>sample.id</code></p>
</td></tr>
<tr valign="top"><td><code>verbose</code></td>
<td>
<p>if <code>TRUE</code>, show information</p>
</td></tr>
</table>


<h3>Details</h3>

<p>&quot;GCTA&quot;: the genetic relationship matrix in GCTA is defined as
G_ij = avg_l [(g_il - 2*p_l)*(g_jl - 2*p_l) / 2*p_l*(1 - p_l)] for
individuals i,j and locus l;
</p>
<p>&quot;Eigenstrat&quot;: the genetic covariance matrix in EIGENSTRAT
G_ij = avg_l [(g_il - 2*p_l)*(g_jl - 2*p_l) / 2*p_l*(1 - p_l)] for
individuals i,j and locus l; the missing genotype is imputed by the dosage
mean of that locus.
</p>
<p>&quot;EIGMIX&quot; / &quot;Weighted&quot;: it is the same as '2 * snpgdsEIGMIX(, ibdmat=TRUE,
diagadj=FALSE)$ibd':
G_ij = [sum_l (g_il - 2*p_l)*(g_jl - 2*p_l)] / [sum_l 2*p_l*(1 - p_l)]
for individuals i,j and locus l;
</p>
<p>&quot;IndivBeta&quot;: it is the same as '2 * snpgdsIndivBeta(, with.id=FALSE)'.
</p>


<h3>Value</h3>

<p>Return a list if <code>with.id = TRUE</code>:
</p>
<table summary="R valueblock">
<tr valign="top"><td><code>sample.id</code></td>
<td>
<p>the sample ids used in the analysis</p>
</td></tr>
<tr valign="top"><td><code>snp.id</code></td>
<td>
<p>the SNP ids used in the analysis</p>
</td></tr>
<tr valign="top"><td><code>grm</code></td>
<td>
<p>the genetic relationship matrix; different methods might
have different meanings and interpretation for estimates</p>
</td></tr>
</table>
<p>If <code>with.id = FALSE</code>, this function returns the genetic relationship
matrix (GRM) without sample and SNP IDs.
</p>


<h3>Author(s)</h3>

<p>Xiuwen Zheng</p>


<h3>References</h3>

<p>Patterson, N., Price, A. L. &amp; Reich, D.
Population structure and eigenanalysis. PLoS Genet. 2, e190 (2006).
</p>
<p>Yang, J., Lee, S. H., Goddard, M. E. &amp; Visscher, P. M.
GCTA: a tool for genome-wide complex trait analysis.
American journal of human genetics 88, 76-82 (2011).
</p>
<p>Zheng X, Weir BS.
Eigenanalysis on SNP Data with an Interpretation of Identity by Descent.
Theoretical Population Biology. 2016 Feb;107:65-76.
doi: 10.1016/j.tpb.2015.09.004
</p>
<p>Weir BS, Zheng X.
SNPs and SNVs in Forensic Science.
Forensic Science International: Genetics Supplement Series. 2015.
doi:10.1016/j.fsigss.2015.09.106
</p>


<h3>See Also</h3>

<p><code><a href="snpgdsPCA.html">snpgdsPCA</a></code>, <code><a href="snpgdsEIGMIX.html">snpgdsEIGMIX</a></code>,
<code><a href="snpgdsIndivBeta.html">snpgdsIndivBeta</a></code>,
<code><a href="snpgdsIndInb.html">snpgdsIndInb</a></code>, <code><a href="snpgdsFst.html">snpgdsFst</a></code>
</p>


<h3>Examples</h3>

<pre>
# open an example dataset (HapMap)
genofile &lt;- snpgdsOpen(snpgdsExampleFileName())


rv &lt;- snpgdsGRM(genofile, method="GCTA")
eig &lt;- eigen(rv$grm)  # Eigen-decomposition

pop &lt;- factor(read.gdsn(index.gdsn(genofile, "sample.annot/pop.group")))
plot(eig$vectors[,1], eig$vectors[,2], col=pop)
legend("topleft", legend=levels(pop), pch=19, col=1:4)


# close the file
snpgdsClose(genofile)
</pre>

<hr /><div style="text-align: center;">[Package <em>SNPRelate</em> version 1.10.2 <a href="00Index.html">Index</a>]</div>
</body></html>
