<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: Create a SNP genotype dataset from a GDS file</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body>

<table width="100%" summary="page for snpgdsCreateGenoSet {SNPRelate}"><tr><td>snpgdsCreateGenoSet {SNPRelate}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>
Create a SNP genotype dataset from a GDS file
</h2>

<h3>Description</h3>

<p>To create a GDS file of genotypes from a specified GDS file.
</p>


<h3>Usage</h3>

<pre>
snpgdsCreateGenoSet(src.fn, dest.fn, sample.id=NULL, snp.id=NULL,
    snpfirstdim=NULL, compress.annotation="ZIP_RA.max", compress.geno="",
    verbose=TRUE)
</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>src.fn</code></td>
<td>
<p>the file name of a specified GDS file</p>
</td></tr>
<tr valign="top"><td><code>dest.fn</code></td>
<td>
<p>the file name of output GDS file</p>
</td></tr>
<tr valign="top"><td><code>sample.id</code></td>
<td>
<p>a vector of sample id specifying selected samples;
if NULL, all samples are used</p>
</td></tr>
<tr valign="top"><td><code>snp.id</code></td>
<td>
<p>a vector of snp id specifying selected SNPs; if NULL,
all SNPs are used</p>
</td></tr>
<tr valign="top"><td><code>snpfirstdim</code></td>
<td>
<p>if TRUE, genotypes are stored in the individual-major
mode, (i.e, list all SNPs for the first individual, and then list all
SNPs for the second individual, etc)</p>
</td></tr>
<tr valign="top"><td><code>compress.annotation</code></td>
<td>
<p>the compression method for the variables except
<code>genotype</code></p>
</td></tr>
<tr valign="top"><td><code>compress.geno</code></td>
<td>
<p>the compression method for the variable
<code>genotype</code></p>
</td></tr>
<tr valign="top"><td><code>verbose</code></td>
<td>
<p>if TRUE, show information</p>
</td></tr>
</table>


<h3>Value</h3>

<p>None.
</p>


<h3>Author(s)</h3>

<p>Xiuwen Zheng</p>


<h3>See Also</h3>

<p><code><a href="snpgdsCreateGeno.html">snpgdsCreateGeno</a></code>, <code><a href="snpgdsCombineGeno.html">snpgdsCombineGeno</a></code>
</p>


<h3>Examples</h3>

<pre>
# open an example dataset (HapMap)
(genofile &lt;- snpgdsOpen(snpgdsExampleFileName()))
# +     [  ] *
# |--+ sample.id    { FStr8 279 ZIP(23.10%) }
# |--+ snp.id   { Int32 9088 ZIP(34.76%) }
# |--+ snp.rs.id    { FStr8 9088 ZIP(42.66%) }
# |--+ snp.position { Int32 9088 ZIP(94.73%) }
# |--+ snp.chromosome   { UInt8 9088 ZIP(0.94%) } *
# |--+ snp.allele   { FStr8 9088 ZIP(14.45%) }
# |--+ genotype { Bit2 9088x279 } *
# |--+ sample.annot [ data.frame ] *
# |  |--+ sample.id { FStr8 279 ZIP(23.10%) }
# |  |--+ family.id { FStr8 279 ZIP(28.37%) }
# |  |--+ geneva.id { Int32 279 ZIP(80.29%) }
# |  |--+ father.id { FStr8 279 ZIP(12.98%) }
# |  |--+ mother.id { FStr8 279 ZIP(12.86%) }
# |  |--+ plate.id  { FStr8 279 ZIP(1.29%) }
# |  |--+ sex   { FStr8 279 ZIP(28.32%) }
# |  |--+ pop.group { FStr8 279 ZIP(7.89%) }

set.seed(1000)
snpset &lt;- unlist(snpgdsLDpruning(genofile))
length(snpset)
# 6547

# close the file
snpgdsClose(genofile)

snpgdsCreateGenoSet(snpgdsExampleFileName(), "test.gds", snp.id=snpset)

####################################################
# check

(gfile &lt;- snpgdsOpen("test.gds"))
# +     [  ] *
# |--+ sample.id    { VStr8 279 ZIP(29.89%) }
# |--+ snp.id   { Int32 6547 ZIP(34.89%) }
# |--+ snp.rs.id    { VStr8 6547 ZIP(40.52%) }
# |--+ snp.position { Int32 6547 ZIP(94.85%) }
# |--+ snp.chromosome   { Int32 6547 ZIP(0.41%) }
# |--+ snp.allele   { VStr8 6547 ZIP(11.51%) }
# |--+ genotype { Bit2 6547x279 } *

# close the file
snpgdsClose(gfile)


unlink("test.gds", force=TRUE)
</pre>

<hr /><div style="text-align: center;">[Package <em>SNPRelate</em> version 1.3.5 <a href="00Index.html">Index</a>]</div>
</body></html>
