<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: Get a common SNP list between two SNP list objects</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body>

<table width="100%" summary="page for snpgdsSNPListIntersect {SNPRelate}"><tr><td>snpgdsSNPListIntersect {SNPRelate}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>
Get a common SNP list between two SNP list objects
</h2>

<h3>Description</h3>

<p>Get a common SNP list by comparing their rs id, chromosome indices and
positions.
</p>


<h3>Usage</h3>

<pre>
snpgdsSNPListIntersect(snplist1, snplist2)
</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>snplist1</code></td>
<td>
<p>the first SNP list object <code><a href="snpgdsSNPListClass.html">snpgdsSNPListClass</a></code></p>
</td></tr>
<tr valign="top"><td><code>snplist2</code></td>
<td>
<p>the second SNP list object <code><a href="snpgdsSNPListClass.html">snpgdsSNPListClass</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Return an object of <code>snpgdsSNPListClass</code> including the following
components:
</p>
<table summary="R valueblock">
<tr valign="top"><td><code>rs.id</code></td>
<td>
<p>SNP id</p>
</td></tr>
<tr valign="top"><td><code>chromosome</code></td>
<td>
<p>SNP chromosome index</p>
</td></tr>
<tr valign="top"><td><code>position</code></td>
<td>
<p>SNP physical position in basepair</p>
</td></tr>
<tr valign="top"><td><code>allele</code></td>
<td>
<p>reference / non-ref alleles from the first SNP list object</p>
</td></tr>
<tr valign="top"><td><code>afreq</code></td>
<td>
<p>allele frequency from the first SNP list object</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Xiuwen Zheng</p>


<h3>See Also</h3>

<p><code><a href="snpgdsSNPList.html">snpgdsSNPList</a></code>, <code><a href="snpgdsSNPListStrand.html">snpgdsSNPListStrand</a></code>
</p>


<h3>Examples</h3>

<pre>
# open an example dataset (HapMap)
genofile &lt;- snpgdsOpen(snpgdsExampleFileName())

# to get a snp list object
snplist1 &lt;- snpgdsSNPList(genofile)
snplist2 &lt;- snpgdsSNPList(genofile)

# a common snp list
snplist &lt;- snpgdsSNPListIntersect(snplist1, snplist2)

summary(snplist$afreq)

# close the file
snpgdsClose(genofile)
</pre>

<hr /><div style="text-align: center;">[Package <em>SNPRelate</em> version 1.3.5 <a href="00Index.html">Index</a>]</div>
</body></html>
